core/interface/static/js/skills.js

```diff
@@ -0,0 +1,202 @@
+$(document).tooltip();
+
+$(function () {
+	let selectedSkillsToDownload = [];
+	let skillStatuses = {};
+
+	function skillStatus(skill) {
+		$.ajax({
+			url: '/skills/checkInstallStatus/',
+			data: {
+				'skill': skill
+			},
+			type: 'POST'
+		}).done(function (status) {
+			skillStatuses[skill] = JSON.stringify(status).trim();
+		}).fail(function () {
+			skillStatuses[skill] = 'failed';
+		});
+	}
+
+	function checkInstallStatus(skill) {
+		skillStatus(skill);
+
+		let status = skillStatuses[skill];
+		if (status == JSON.stringify('installed')) {
+			$('#' + skill + 'InstallTile').remove();
+		} else if (status == JSON.stringify('failed') || status == JSON.stringify('unknown')) {
+			$('#' + skill + 'InstallTile').children('.skillStoreSkillWaitAnimation').hide();
+			$('#' + skill + 'InstallTile').children('.skillStoreSkillDownloadFail').css('display', 'flex');
+		} else {
+			setTimeout(function () {
+				checkInstallStatus(skill);
+			}, 5000);
+		}
+	}
+
+	function checkUpdateStatus(skill, $skillContainer) {
+		skillStatus(skill);
+		let status = skillStatuses[skill];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable Assigned to Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4699682)